classes/submission.class.php

```diff
@@ -1328,4 +1359,34 @@ public function gettiiretrytime() {
     public function settiiretrytime($tiiretrytime) {
         $this->tiiretrytime = $tiiretrytime;
     }
+
+    /**
+     * Reset retries.
+     */
+    private function reset_retries() {
+        $this->settiiattempts(0);
+        $this->settiiretrytime(0);
+    }
+
+    /**
+     * @param $error string Error code from Turnitin.
+     * @param $retry_attempts int Maximum retry attempts.
+     */
+    private function set_error_with_max_retry_attempts($error, $retry_attempts) {
```

Comment:
Can we add a description for this method and below? Moodle code checker will complain.

Reply:
As it is a private method so do moodle still show warning?

Reply:
I think so. You can install the Moodle PHPDoc checker if you want to check. There's also the Moodle PHP code checker which checks the code itself. This one is bundled with our Moodle docker but usually requires modifying one line inside it to work.